### ISSUE TITLE
ES6 Usage / Fixes / NPM etc..

### ISSUE THREAD

**Issue by MarketingPip** [live-codes/go2js#2]
Would be great if you could fix out some kinks so this can be used easily in browser like so...

```js
import { compile, format } from 'https://esm.sh/go2js@0.2.0';



const code = `
package main
import ("syscall/js"
"fmt")
func main() {fmt.Println("Hello, from Go")
js.Global().Call("alert", "Hello, JavaScript")
println("Hello, JS console")}
`;

format(code)
  .then((formatted) => {
    console.log(formatted);
    // return compile(formatted, 'https://unpkg.com/go2js/build/');
    return compile(formatted);
  })
  .then(eval)
  .catch(console.warn);
```

Currently doesn't work. Possibly you can set base name to resolve from in the module etc?

**Comment by hatemhosny** [comment]
You are correct. 
Thank you for letting me know. 

The correct file is https://unpkg.com/@live-codes/go2js@0.5.0/build/index.js

You can try it like this for now, till I fix it
```js
import { compile, format } from 'https://unpkg.com/@live-codes/go2js@0.5.0/build/index.js';

const code = `
package main
import ("syscall/js"
"fmt")
func main() {fmt.Println("Hello, from Go")
js.Global().Call("alert", "Hello, JavaScript")
println("Hello, JS console")}
`;

format(code)
  .then((formatted) => {
    console.log(formatted);
    return compile(formatted);
  })
  .then(eval)
  .catch(console.warn);
```
[Try in LiveCodes](https://livecodes.io/?x=id/7dz7aaa5ryt) 

Thank you.

**Comment by MarketingPip** [comment]
@hatemhosny - you are the best! 

Hope all is well - long time no speak / see! Hahaha! Still hoping we can collab on something etc... Feel free to reach me via LinkedIn - you can find via my profile!

ps; I can not seem to resolve the promise even when awaited for compiled code? (Nevermind - dumb its just fetching lol)

**Comment by hatemhosny** [comment]
Nice to hear from you. 
Thank you for reaching out and for reporting the issue. 

> ps; I can not seem to resolve the promise even when awaited for compiled code? (Nevermind - dumb its just fetching lol)

No worries. It does take some time to fetch and compile.

I'll fix the npm package configuration soon once I have some time.

**Comment by MarketingPip** [comment]
@hatemhosny - got a working demo here. (unpkg seems terribly slow lol)

```js
import { compile, format } from 'https://cdn.jsdelivr.net/npm/@live-codes/go2js@0.5.0/build/index.min.js';

const code = `
package main
import ("syscall/js"
"fmt")
func main() {fmt.Println("Hello, from Go")
 
println("Hello, JS console")}
`;

 
 async function processCode(code) {
  try {
    const formatted = await format(code);
   
 
    
   const results = await compile(formatted, `https://cdn.jsdelivr.net/npm/@live-codes/go2js@0.5.0/build/`);
    
    console.log(results)
    eval(results);
  } catch (error) {
    console.warn(error);
  }
}

await processCode(code)
```

**Comment by hatemhosny** [comment]
This issue has been fixed in v0.5.3

this code should now work with no problems

```ts
import { compile, format } from 'https://cdn.jsdelivr.net/npm/@live-codes/go2js';

const code = `
package main
import ("syscall/js"
"fmt")
func main() {fmt.Println("Hello, from Go")
js.Global().Call("alert", "Hello, JavaScript")
println("Hello, JS console")}
`;

format(code)
  .then((formatted) => {
    console.log(formatted);
    return compile(formatted);
  })
  .then(eval)
  .catch(console.warn);
```

[Open in LiveCodes](https://livecodes.io/?x=id/si989jz53ed)
